dxr/filters.py

```diff
@@ -66,6 +67,13 @@ def __init__(self, term, enabled_plugins):
         self._term = term
         self._enabled_plugins = enabled_plugins
 
+    def __str__(self):
+        """Return a string representation of this Filter as name:term, as it
+        appeared in the query string.
+        """
+
+        return "%s:%s" % (self.name, self._term['arg'])
```

Comment:
Like the JS, this is still naive. What if the term has a space in it? You basically have to do the reverse of the query parser—at least as far as reconstructing a single term.